geonode/geoserver/management/commands/sync_geonode_layers.py

```diff
@@ -43,10 +43,10 @@ def sync_geonode_layers(ignore_errors, filter, username):
             print 'Syncing layer %s/%s: %s' % (count, layers_count, layer.name)
             # sync permissions in GeoFence
             perm_spec = json.loads(_perms_info_json(layer))
+            # re-sync GeoFence security rules
             layer.set_permissions(perm_spec)
             # recalculate the layer statistics
             set_attributes_from_geoserver(layer, overwrite=True)
-            layer.save()
```

Comment:
This command is not only related to permissions, but to general synchronization between GeoServer/GeoFence/Django. I believe without this layer.save() thumbnails and styles won't be synced

Reply:
@capooti this was my intention. I mean, it seems to me dangerous and also a waste of time to regenerate all the thumbnails for the layers synced. It seems to me that the thumbnail, except the first time, should be manually updated by the user from the gui or at most we should envisage a specific management command for this.

Reply:
well this command was thought as a sort of updatelayers alternative version which doesn't use GetCapabilities (which can be broken and it is broken and really slow in our case). The use case for us was to use it as a replacement to it.
We are also using it to migrate GeoNode itself, which explains why we want to update the thumbnails - in fact they are being generated for the first time in such a case. How about to add it as a command option that launch layer.save() if enabled?

Reply:
@capooti yes, good point. Make sense to add an explicit option which will warn users to eventually provide a filter whenever they want to explicitly update the thumbnails too. I will push the change in a few unless there are other objections.

Reply:
@afabiani I would add the following optional arguments: --updatepermissions, --updatethumbnails
If none are set all actions will be executed, otherwise only the ones set are executed.

Reply:
great! thanks @afabiani and @giohappy